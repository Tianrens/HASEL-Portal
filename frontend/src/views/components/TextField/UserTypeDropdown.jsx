import { React, useEffect } from 'react';
import { TextField, MenuItem } from '@mui/material';
import styles from './TextField.module.scss';
import selectMenuProps from '../../../assets/selectMenuProps';
import { NON_ADMIN_ACCOUNT_TYPE, ALL_ACCOUNT_TYPE } from '../../../config/accountTypes';

const UserTypeDropdown = ({ adminView, required, initialValue = '', setValue, ...props }) => {
    const types = adminView ? ALL_ACCOUNT_TYPE : NON_ADMIN_ACCOUNT_TYPE;
    // eslint-disable-next-line react-hooks/exhaustive-deps
    useEffect(() => setValue(initialValue), [initialValue]);

    return (
        <div className={styles.container}>
            <p className={styles.title}>
                Account Type
                {required && <span className={styles.asterisk}> *</span>}
            </p>

            <TextField
                title='Account Type'
                select
                required={required}
                defaultValue={initialValue}
                SelectProps={{ MenuProps: selectMenuProps }}
                onChange={(event) => setValue(event.target.value)}
                {...props}
            >
                {Object.keys(types).map((option) => (
                    <MenuItem key={option} value={option}>
                        {types[option]}
                    </MenuItem>
                ))}
            </TextField>
        </div>
    );
};
export default UserTypeDropdown;
