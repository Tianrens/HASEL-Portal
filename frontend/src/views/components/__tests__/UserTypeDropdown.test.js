import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import UserTypeDropdown from '../TextField/UserTypeDropdown';

test('UserTypeDropdown renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <UserTypeDropdown setValue={() => {}} />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
