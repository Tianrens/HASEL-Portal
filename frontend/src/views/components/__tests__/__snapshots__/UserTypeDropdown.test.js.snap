// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTypeDropdown renders properly 1`] = `
<div
  className="container"
>
  <p
    className="title"
  >
    Account Type
  </p>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    title="Account Type"
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        autoFocus={false}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        disabled={false}
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1a1ihhj"
        >
          <span
            className="notranslate"
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
